lib/avromatic.rb

```diff
@@ -4,13 +4,14 @@
 require 'avromatic/model_registry'
 require 'avromatic/messaging'
 require 'active_support/core_ext/string/inflections'
+require 'avromatic/patches'
 
 module Avromatic
   class << self
     attr_accessor :schema_registry, :registry_url, :schema_store, :logger,
                   :messaging, :type_registry, :nested_models,
                   :use_custom_datum_reader, :use_custom_datum_writer,
-                  :use_schema_fingerprint_lookup
```

Comment:
use_encoding_providers actually isn't intended to be configurable or even user visible. Here's whats going on: in order for avromatic to use encoding providers (and thus cache the avro encodings for immutable models) two things need to be true: 
- use_custom_datum_writer must be true because caching is actually implemented, in part, in the custom datum writer.
- the avro-patches gem must be present because, with this PR, the caching implementation depends on the schema validation patches it provides (which we patch further).

Because avromatic doesn't actually require avro-patches and use_custom_datum_writer is user configurable, we need to check both of these conditions during serialization to figure out whether we can include encoding providers in the output.

I added Avromatic.use_encoding_providers just to avoid having to explicitly check for the presence of avro-patches deep in the serialization code. (raw_serialization.rb:32)

Reply:
Makes sense. Can we hide this setting somewhere else so it isn't confused with the other settings here that are user configurable?